Sources/Sentry/SentryTracer.m

```diff
@@ -335,6 +424,19 @@ - (void)captureTransaction
         additionalEnvelopeItems:additionalEnvelopeItems];
 }
 
+- (void)trimEndTimestamp
+{
+    NSArray<id<SentrySpan>> *sortedChildrenNewestFirst =
+        [_children sortedArrayUsingComparator:^NSComparisonResult(id<SentrySpan> first,
+            id<SentrySpan> second) { return [first.timestamp compare:second.timestamp]; }];
+
+    // Only trim if all children are finished. Otherwise the timestamp would be nil
+    NSDate *timestamp = sortedChildrenNewestFirst.lastObject.timestamp;
+    if (timestamp) {
+        self.timestamp = timestamp;
+    }
```

Comment:
picking: Sorting is at least O(n log n), this could be solve with O(n) 😉

Reply:
Good catch, thanks.